src/parsec_client/operations/attest_key.md

```diff
@@ -0,0 +1,111 @@
+# AttestKey
+
+**(EXPERIMENTAL)** Attest that a Parsec-managed key is protected by a hardware backend. Opcode: 30
+(`0x001E`)
+
+## Parameters
+
+| Name                 | Type                                                    | Description                               |
+|----------------------|---------------------------------------------------------|-------------------------------------------|
+| `attested_key_name`  | String                                                  | Name of the key to attest                 |
+| `parameters`         | [`AttestationMechanismParams`](#attestation-mechanisms) | Attestation mechanism-specific parameters |
+| `attesting_key_name` | String                                                  | Name of the key to use for attesting      |
+
+- The exact usage flags required by the attesting key are determined by the mechanism used, as
+   described below.
+
+## Results
+
+| Name     | Type                                           | Description                           |
+|----------|------------------------------------------------|---------------------------------------|
+| `output` | [`AttestationOutput`](#attestation-mechanisms) | Attestation mechanism-specific output |
+
+## Specific response status codes
+
+TBD
+
+## Description
+
+This operation performs a key attestation using a mechanism supported by the backend holding the
+key. The purpose of the operation is to help a Parsec client provide proof to a 3rd party that some
+key provisioned by the client is indeed stored and secured by a hardware backend. As such, the
+operation is backed by native functionality in the hardware to attest to ownership of the key.
+
+Given the wide variety of possible mechanisms, many of the properties, restrictions, and formats
+involved are mechanism-dependent, including:
+
+- Properties of the attested key (e.g. whether it was created within the backend or imported)
+- Properties of the attesting key (e.g. if it must be able to sign or decrypt)
+- Number, content, and purpose of parameters required by the attestation
+- Contents and format of the output
+- Whether or not the attesting key can be specified
+
+All such characteristics are thoroughly described per mechanism in the section below.
+
+All instances of backends that support `AttestKey` must be configured with a default, root key that
+has been pre-provisioned and which can be used to produce attestations. This default attesting key
+is selected by leaving the `attesting_key_name` empty. If the backend allows other keys to be used
+for attesting, attestation chains can be created starting from the root key.
+
+`AttestKey` applies to asymmetric key pairs only.
+
+## Attestation mechanisms
```

Comment:
For the attestation mechanisms types, since it seems to be an enumerated type, it might be good to follow the same idea than for the [`KeyType`](https://parallaxsecond.github.io/parsec-book/parsec_client/operations/psa_key_attributes.html#keytype-type) type:
- separate it in its own file
- describe each `AttestationMechanismParams` and `AttestationOutput` as enumerated types: "A AttestationMechanismParams type can contain one of the following attestation mechanism: * ActivateCredential * ..."
- have a subsection where the ActivateCredential type is explained as it is done here but doing it in high-level term, without specifying the contract. Similarly as what we do for `EccKeyPair` with "Uses one of the curve family supported".

Reply:
I agree with the first two points, I was considering doing it like that, but wasn't sure about adding another file. 

> doing it in high-level term, without specifying the contract.

Not sure I follow what this means. Not to describe how the operation works and only focus on the formats/expected outputs?

Reply:
> Not sure I follow what this means. Not to describe how the operation works and only focus on the formats/expected outputs?

I was comparing it with some of our enumerated types where we don't exactly show the contract but only "Uses one of the ... supported". And let the implementors of those contracts in the various languages (currently only protobuf) interpret that how they want. But I guess in the case of those types, the message can be written formaly as you did because the parameters are not just enums.

Reply:
I guess that's the reason - for those types you can say "it can be one of these values", but for `ActivateCredential` the contents are more closely resembling `PsaExportPublicKey`, so we have to define what the format of the data is, you can't leave it to the different contracts. 

I guess my question is whether the big description of the mechanism is useful in this document?

Reply:
> I guess my question is whether the big description of the mechanism is useful in this document?

If we were to put the mechanisms in a separate document, I think it would make sense to be included there.